File: voidr-playwright/page_objects/home_page.py
import random

from page_objects.base_page import BasePage

class HomePage(BasePage):
    def __init__(self, page):
        super().__init__(page)
        self.cart_badge = 'span[class="shopping_cart_badge"]'

    def addItemToCart(self):
        addToCartButtons = self.page.query_selector_all('button[id^="add-to-cart-"]')

        if addToCartButtons:
            randomButton = random.choice(addToCartButtons)
            randomButton.click()
        else:
            raise Exception('No "Add to Cart" button was found.')

    def removeItemFromCart(self):
        removeFromCartButtons = self.page.query_selector_all('button[id^="remove-"]')

        if removeFromCartButtons:
            randomButton = random.choice(removeFromCartButtons)
            randomButton.click()
        else:
            raise Exception('No "Remove" button was found')
    def goToCart(self):
        self.page.goto('https://www.saucedemo.com/cart.html')

    def getCartCount(self):
        return self.page.text_content(self.cart_badge)

    def logout(self):
        self.page.click('button[id="react-burger-menu-btn"]')
        self.page.click('a[id="logout_sidebar_link"]')

File: playwright-js-voidr/tests/page-objects/home-page.js
// homePage.js
const BasePage = require('./base-page'); // Certifique-se de que o caminho está correto

class HomePage extends BasePage {
  constructor(page) {
    super(page);
    this.cartBadge = 'span[class="shopping_cart_badge"]';
  }

  async addItemToCart() {
    await this._clickRandomButton('button[id^="add-to-cart-"]', 'No "Add to Cart" button was found.');
  }

  async removeItemFromCart() {
    await this._clickRandomButton('button[id^="remove-"]', 'No "Remove" button was found.');
  }

  async _clickRandomButton(selector, errorMessage) {
    const buttons = await this.page.$$(selector);
    if (buttons.length > 0) {
      const randomIndex = Math.floor(Math.random() * buttons.length);
      await buttons[randomIndex].click();
    } else {
      throw new Error(errorMessage);
    }
  }

  async goToCart() {
    await this.page.goto('https://www.saucedemo.com/cart.html');
  }

  async getCartCount() {
    return await this.page.textContent(this.cartBadge);
  }

  async logout() {
    await this.page.click('button[id="react-burger-menu-btn"]');
    await this.page.click('a[id="logout_sidebar_link"]');
  }
}

module.exports = HomePage;
